streams/api/src/main/java/org/eclipse/microprofile/reactive/streams/ProcessorBuilder.java

```diff
@@ -66,14 +66,14 @@
   }
 
   /**
-   * Returns a stream containing all the elements from this stream, 
-   * additionaly perfoming the provided action on each element.
+   * Returns a stream containing all the elements from this stream, additionally performing the provided action on each
+   * element.
    *
    * @param consumer The function called for every element.
-   * @return A new processor builder that consumes elements of type <code>T</code> and emits the same elements. In between, the given function 
-   * is called for each element.
+   * @return A new processor builder that consumes elements of type <code>T</code> and emits the same elements. In between,
+   * the given function is called for each element.
    */
-  public ProcessorBuilder<T, R> peek(Consumer<? super R> consumer) {
+  public ProcessorBuilder<T, T> peek(Consumer<? super T> consumer) {
```

Comment:
I'm pretty sure this should be `<T, R>`, and `R` should be the peeked type. A processor takes in elements of type `T`, and emits elements of type `R`. This allows peeking on what is emitted, so the result is still something of type `R`.

Reply:
Yes, you are right, I misread the generics.